contracts/JBETHERC20SplitsPayer.sol

```diff
@@ -0,0 +1,412 @@
+    if (_domain != defaultSplitsDomain) defaultSplitsDomain = _domain;
+    if (_group != defaultSplitsGroup) defaultSplitsGroup = _group;
+
+    emit SetDefaultSplits(_projectId, _domain, _group, msg.sender);
+  }
+
+  /** 
+    @notice 
+    Make a payment to the specified project after first spliting the amount among the saved splits.
+
+    @dev
+    Set the `payer` as this contract's address if it is to manage the token transfer from the msg.sender to the destination terminal.
+    
+    @param _projectId The ID of the project that is being paid after.
+    @param _token The token being paid in.
+    @param _payer The address from whom the payment is originating. Set the `payer` as this contract's address if it is to manage the token transfer from the msg.sender to the destination terminal.
+    @param _amount The amount of tokens being paid, as a fixed point number. If this terminal's token is ETH, this is ignored and msg.value is used in its place.
+    @param _decimals The number of decimals in the `_amount` fixed point number. If this terminal's token is ETH, this is ignored and 18 is used in its place, which corresponds to the amount of decimals expected in msg.value.
+    @param _beneficiary The address who will receive tokens from the payment made with leftover funds.
+    @param _minReturnedTokens The minimum number of project tokens expected in return, as a fixed point number with 18 decimals.
+    @param _preferClaimedTokens A flag indicating whether the request prefers to mint project tokens into the beneficiaries wallet rather than leaving them unclaimed. This is only possible if the project has an attached token contract. Leaving them unclaimed saves gas.
+    @param _memo A memo to pass along to the emitted event, and passed along the the funding cycle's data source and delegate.  A data source can alter the memo before emitting in the event and forwarding to the delegate.
+    @param _metadata Bytes to send along to the data source and delegate, if provided.
+  */
+  function pay(
+    uint256 _projectId,
+    address _token,
+    address _payer,
+    uint256 _amount,
+    uint256 _decimals,
+    address _beneficiary,
+    uint256 _minReturnedTokens,
+    bool _preferClaimedTokens,
+    string memory _memo,
+    bytes memory _metadata
+  ) public payable virtual override {
+    // ETH shouldn't be sent if this terminal's token isn't ETH.
+    if (address(_token) != JBTokens.ETH) {
+      if (msg.value > 0) revert NO_MSG_VALUE_ALLOWED();
+      // Transfer tokens to this terminal from the msg sender.
+      if (_payer == address(this))
+        IERC20(_token).transferFrom(msg.sender, payable(address(this)), _amount);
+    } else {
+      _payer = address(this);
+      _amount = msg.value;
+      _decimals = 18;
+    }
+
+    // Route the payment to the splits.
+    // Pay the split and get a reference to the amount paid.
+    uint256 _leftoverAmount = _payToSplits(
+      defaultProjectId,
```

Comment:
`defaultSplitsProjectId` instead? (defaultProjectId is inherited from project payer)

Reply:
Leave it as is. The `defaultSplitsProjectId` is just used to index the project to which the splits belong. This should, in most practical cases I can think of, be the same as the `defaultProjectId` inherited. But deployers may wanna customize the payout of leftover some other way.